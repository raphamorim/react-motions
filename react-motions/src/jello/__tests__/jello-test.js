import React from 'react';
import { mount } from 'enzyme';
import Jello from '../jello';

describe('Jello', () => {
  it('should render', () => {
    const wrapper = shallow(<Jello>Jello</Jello>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<Jello duration={3}>Jello</Jello>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<Jello fillMode="both">Jello</Jello>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should receive transformOrigin prop', () => {
    const wrapper = mount(<Jello transformOrigin="center">Jello</Jello>);
    expect(wrapper.props().transformOrigin).toEqual("center");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<Jello infinite>Jello</Jello>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
