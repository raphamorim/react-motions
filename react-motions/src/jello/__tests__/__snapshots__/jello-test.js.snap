// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jello should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jello
    duration={1}
    fillMode="both"
    infinite={false}
    transformOrigin="center"
  >
    Jello
  </Jello>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Jello",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "jello",
        "WebkitTransformOrigin": "center",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "jello",
        "transformOrigin": "center",
      },
    },
    "ref": null,
    "rendered": "Jello",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Jello",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "jello",
          "WebkitTransformOrigin": "center",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "jello",
          "transformOrigin": "center",
        },
      },
      "ref": null,
      "rendered": "Jello",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
